### PR TITLE
Refactor the code to make task functions simple functions

### PR DESCRIPTION
**Pull request by sdispater**
This PR makes the `manager.task()` method a true decorator by not altering the decorated function  (except the addition of a `defer()` method to it to replicate the previous behavior).

What this allows is the ability to start decorating existing functions without modifying their signature (no `task_run` anymore) and leave them unit testable without relying on the internals of `cabbage`.

This code came up when trying to address #3 and I think if we go the route of this PR this makes the implementation of test specific tools in `cabbage` mostly unnecessary.

To sum it up, we can now have something like this:

```python
from cabbage import TaskManager

manager = TaskManager()


@manager.task(queue="default")
def add(a, b):
    return a + b


add(1, 2)  # 3
```

### CODE REVIEW

File: cabbage/task_worker.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def run(self, timeout: int = SOCKET_TIMEOUT) -> None:
                 break
 
             logger.debug("Waiting")
-            select.select(rlist=[connection], wlist=[], xlist=[], timeout=timeout)
+            select.select([connection], [], [], timeout)
```
Comment (sdispater) [inline cabbage/task_worker.py]: For information, I had to change this because this causes an error (not present in the tests) `TypeError: select() takes no keyword arguments`

Comment (ewjoachim) [reply cabbage/task_worker.py]: Oo ok, sorry about that :/

File: cabbage/tasks.py [hunk 2]
```diff
@@ -9,19 +9,17 @@
...
-        self.func: Optional[Callable] = None
-        self.name = name
-
-    def __call__(self, func: Callable) -> "Task":
-        self.func = func
-        if not self.name:
-            self.name = func.__name__
-        self.manager.register(self)
-        return self
+        self.func: Optional[Callable] = func
```
Comment (ewjoachim) [inline cabbage/tasks.py]: It's not optional anymore, then, is it ?

File: cabbage/task_worker.py (hunk 1 above)
Comment (sdispater) [reply cabbage/task_worker.py]: This weren't triggered by the tests so there was no way to know

File: cabbage/tasks.py (hunk 2 above)
Comment (sdispater) [reply cabbage/tasks.py]: That's true. I will update it

File: cabbage/task_worker.py (hunk 1 above)
Comment (ewjoachim) [reply cabbage/task_worker.py]: Well, it tells something about our tests, doesn't it ? Can we update the acceptance test to do something ? Either defering a task from another task or spinning a thread that waits for a second and then creates a task should do the trick.

Comment (sdispater) [reply cabbage/task_worker.py]: I updated the acceptance tests to cover this (by using threads). It fails as expected with the call to `select()` we previously did.

File: cabbage/tasks.py (hunk 2 above)
Comment (sdispater) [reply cabbage/tasks.py]: Updated :-)

**Comment by ewjoachim** [comment]
@mgu @k4nar I'd like a second opinion

**Comment by ewjoachim** [comment]
Ok, let's merge it then, and we'll see later how we want to access current task info.

Maybe we will add a parameter to the task decorator to decide if we want to pass this or not... In a later PR :D

**Comment by ewjoachim** [comment]
Hoping it's ok with everyone, I'll merge now, because I'd like to have a go at some tickets today.
